### PR TITLE
[INLONG-7166][DataProxy] Fix audit data reporting

### LINKED ISSUE #7166: [Bug][DataProxy] Fix audit data report
### What happened

No audit data report

### What you expected to happen

Fix it

### How to reproduce

Run process

### Environment

_No response_

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fixes #7166

### Modifications

- add audit report for new mq sink
- fix concurrency issue on config operation

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java [hunk 1]
```diff
@@ -193,6 +195,9 @@ public void onCompletion(RecordMetadata arg0, Exception ex) {
                     sinkContext.addSendResultMetric(event, topic, true, sendTime);
                     sinkContext.getDispatchQueue().release(event.getSize());
                     event.ack();
+                    for (ProxyEvent auditEvent : event.getEvents()) {
+                        AuditUtils.add(AuditUtils.AUDIT_ID_DATAPROXY_SEND_SUCCESS, auditEvent);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java:199]: Repeat audit report.
https://github.com/apache/inlong/blob/master/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java
image

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java [hunk 2]
```diff
@@ -238,6 +243,7 @@ public void onCompletion(RecordMetadata arg0, Exception ex) {
                     sinkContext.addSendResultMetric(event, topic, true, sendTime);
                     sinkContext.getDispatchQueue().release(event.getSize());
                     event.ack();
+                    AuditUtils.add(AuditUtils.AUDIT_ID_DATAPROXY_SEND_SUCCESS, event.getSimpleProfile());
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java:246]: Repeat audit report.
https://github.com/apache/inlong/blob/master/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java
image

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java [hunk 3]
```diff
@@ -280,6 +286,9 @@ public void onCompletion(RecordMetadata arg0, Exception ex) {
                     sinkContext.addSendResultMetric(event, topic, true, sendTime);
                     sinkContext.getDispatchQueue().release(event.getSize());
                     event.ack();
+                    for (ProxyEvent auditEvent : event.getEvents()) {
+                        AuditUtils.add(AuditUtils.AUDIT_ID_DATAPROXY_SEND_SUCCESS, auditEvent);
+                    }
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/kafka/KafkaHandler.java:291]: Repeat audit report.
https://github.com/apache/inlong/blob/master/inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java
image

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**